amaranth-stubs/lib/data.pyi

```diff
@@ -42,8 +42,8 @@ class Field:
 
 class Layout(ShapeCastable[View[Self]], metaclass=ABCMeta):
     @staticmethod
-    def cast(obj: ShapeCastable) -> Layout:
-        """Cast a shape-castable object to a layout."""
+    def cast(obj: ShapeLike) -> Layout:
+        """Cast a shape-like object to a layout."""
```

Comment:
This is unnecessarily loose. Only sensible arguments for `cast` are `ShapeCastable`s, otherwise an error is raised.